samples/SocketsSample/wwwroot/dynamic.html

```diff
@@ -0,0 +1,152 @@
+        <input type="button" id="disconnect" value="Disconnect" />
+    </div>
+
+
+    <h4>To Everybody</h4>
+    <form class="form-inline">
+        <div class="input-append">
+            <input type="text" id="message-text" placeholder="Type a message, name or group" />
+            <input type="button" id="broadcast" class="btn" value="Broadcast" />
+            <input type="button" id="broadcast-exceptme" class="btn" value="Broadcast (All Except Me)" />
+            <input type="button" id="join" class="btn" value="Enter Name" />
+            <input type="button" id="join-group" class="btn" value="Join Group" />
+            <input type="button" id="leave-group" class="btn" value="Leave Group" />
+        </div>
+    </form>
+
+    <h4>To Me</h4>
+    <form class="form-inline">
+        <div class="input-append">
+            <input type="text" id="me-message-text" placeholder="Type a message" />
+            <input type="button" id="send" class="btn" value="Send to me" />
+        </div>
+    </form>
+
+    <h4>Private Message</h4>
+    <form class="form-inline">
+        <div class="input-prepend input-append">
+            <input type="text" name="private-message" id="private-message-text" placeholder="Type a message" />
+            <input type="text" name="user" id="target" placeholder="Type a user or group name" />
+
+            <input type="button" id="privatemsg" class="btn" value="Send to user" />
+            <input type="button" id="groupmsg" class="btn" value="Send to group" />
+        </div>
+    </form>
+
+    <ul id="message-list"></ul>
+</body>
+</html>
+<script src="lib/signalr-client/signalr-client.js"></script>
+<script src="utils.js"></script>
+<script>
+var isConnected = false;
+function invoke(connection, method) {
+    if (!isConnected) {
+        return;
+    }
+    var argsArray = Array.prototype.slice.call(arguments);
+    connection.invoke.apply(connection, argsArray.slice(1))
+            .then(result => {
+                console.log("invocation completed successfully: " + (result === null ? '(null)' : result));
+
+                if (result) {
+                    addLine('message-list', result);
+                }
+            })
+            .catch(err => {
+                addLine('message-list', err, 'red');
+            });
+}
+
+function getText(id) {
+    return document.getElementById(id).value;
+}
+
+let transportType = signalR.TransportType[getParameterByName('transport')] || signalR.TransportType.WebSockets;
+
+document.getElementById('head1').innerHTML = signalR.TransportType[transportType];
+
+let connectButton = document.getElementById('connect');
+let disconnectButton = document.getElementById('disconnect');
+disconnectButton.disabled = true;
+var connection;
+
+click('connect', event => {
+    connectButton.disabled = true;
+    disconnectButton.disabled = false;
+    let http = new signalR.HttpConnection(`http://${document.location.host}/dynamic`, { transport: transportType });
```

Comment:
Is it the only meaningful difference from hubs.html? If so, I would recommend just passing hub type as a parameter and reuse the code we already have and avoid code duplication.